diff --color -ru uoscore-uedhoc/samples/linux_edhoc_oscore/responder_server/Makefile uoscore-uedhoc.patched/samples/linux_edhoc_oscore/responder_server/Makefile
--- uoscore-uedhoc/samples/linux_edhoc_oscore/responder_server/Makefile	2023-09-15 00:08:22.287094000 +0300
+++ uoscore-uedhoc.patched/samples/linux_edhoc_oscore/responder_server/Makefile	2023-09-15 00:17:53.166814300 +0300
@@ -153,7 +153,7 @@
 	$(CXX) -c $(CXXFLAGS) $< -o $@
 
 $(BUILD_DIR)/%.o: %.c Makefile | $(BUILD_DIR) 
-	$(CC) -c $(CFLAGS) -Wa,-a,-ad,-alms=$(BUILD_DIR)/$(notdir $(<:.c=.lst)) $< -o $@
+	$(CC) -c $(CFLAGS) $< -o $@
 
 $(BUILD_DIR)/$(TARGET): $(OBJECTS) Makefile $(USOCORE_UEDHOC_PATH)/Makefile $(USOCORE_UEDHOC_OBJ)
 	$(MAKE) -C $(USOCORE_UEDHOC_PATH)
