diff --color -ru dtls-fuzzer-usenix20-artifact/src/main/java/se/uu/it/dtlsfuzzer/execute/AbstractInputExecutor.java dtls-fuzzer-usenix20-artifact.patched/src/main/java/se/uu/it/dtlsfuzzer/execute/AbstractInputExecutor.java
--- dtls-fuzzer-usenix20-artifact/src/main/java/se/uu/it/dtlsfuzzer/execute/AbstractInputExecutor.java	2020-06-28 14:23:57.000000000 +0200
+++ dtls-fuzzer-usenix20-artifact.patched/src/main/java/se/uu/it/dtlsfuzzer/execute/AbstractInputExecutor.java	2023-06-20 16:00:15.679963116 +0200
@@ -13,6 +13,14 @@
 import se.uu.it.dtlsfuzzer.sut.io.TlsInput;
 import se.uu.it.dtlsfuzzer.sut.io.TlsOutput;
 
+import java.io.FileOutputStream;
+import java.io.File;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
 public abstract class AbstractInputExecutor {
 	private static final Logger LOGGER = LogManager
 			.getLogger(AbstractInputExecutor.class.getName());
@@ -118,6 +126,35 @@
 				// Then we check for repeating outputs (some implementations can
 				// generate a non-deterministic number of messages)"
 				ProtocolMessage m = action.getReceivedMessages().get(i);
+				// -------------------------
+				// DUMP THE RECEIVED MESSAGE
+				try {
+					File fileReader = new File("recv.length");
+					int recordLength = 0;
+					if(fileReader.exists()) {
+						Scanner scanner = new Scanner(fileReader);
+						recordLength = scanner.nextInt();
+					}
+					recordLength += 1;
+					FileWriter fileWriter = new FileWriter("recv.length");
+					PrintWriter printWriter = new PrintWriter(fileWriter);
+					printWriter.print(recordLength+"\n");
+					fileWriter.close();
+					printWriter.close();
+
+					byte[] val = m.getCompleteResultingMessage().getOriginalValue();
+					byte[] len = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(val.length).array();
+					FileOutputStream fosRep = new FileOutputStream("recv.replay",true);
+					fosRep.write(len);
+					fosRep.write(val);
+					fosRep.close();
+					FileOutputStream fosRaw = new FileOutputStream("recv.raw",true);
+					fosRaw.write(val);
+					fosRaw.close();
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+				// -------------------------
 				if (lastSeen != null && lastSeen.equals(m.toCompactString())) {
 					if (!skipStar) {
 						// insert before ,
diff --color -ru dtls-fuzzer-usenix20-artifact/src/main/java/se/uu/it/dtlsfuzzer/execute/ExecuteInputHelper.java dtls-fuzzer-usenix20-artifact.patched/src/main/java/se/uu/it/dtlsfuzzer/execute/ExecuteInputHelper.java
--- dtls-fuzzer-usenix20-artifact/src/main/java/se/uu/it/dtlsfuzzer/execute/ExecuteInputHelper.java	2020-06-28 14:23:57.000000000 +0200
+++ dtls-fuzzer-usenix20-artifact.patched/src/main/java/se/uu/it/dtlsfuzzer/execute/ExecuteInputHelper.java	2023-06-20 15:48:40.993313492 +0200
@@ -1,6 +1,13 @@
 package se.uu.it.dtlsfuzzer.execute;
 
 import java.io.IOException;
+import java.io.FileOutputStream;
+import java.io.File;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
 import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
@@ -79,6 +86,36 @@
 	public final void sendRecords(List<AbstractRecord> records, State state) {
 		SendMessageHelper helper = new SendMessageHelper();
 		try {
+			// ---------------------
+			// DUMP THE SENT MESSAGE
+			//LOGGER.info("records.size()=" + records.size());
+      File fileReader = new File("send.length");
+      int recordLength = 0;
+      if(fileReader.exists()) {
+        Scanner scanner = new Scanner(fileReader);
+        recordLength = scanner.nextInt();
+      }
+      recordLength += records.size();
+      //LOGGER.info("recordLength=" + recordLength);
+      FileWriter fileWriter = new FileWriter("send.length");
+      PrintWriter printWriter = new PrintWriter(fileWriter);
+      printWriter.print(recordLength+"\n");
+      fileWriter.close();
+      printWriter.close();
+
+      for (AbstractRecord record : records) {
+        byte[] val = record.getRecordSerializer().serialize();
+        byte[] len = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(val.length).array();
+        FileOutputStream fosRep = new FileOutputStream("send.replay",true);
+        fosRep.write(len);
+        fosRep.write(val);
+        fosRep.close();
+        FileOutputStream fosRaw = new FileOutputStream("send.raw",true);
+        fosRaw.write(val);
+        fosRaw.close();
+      }
+			// ---------------------
+
 			helper.sendRecords(records, state.getTlsContext());
 		} catch (IOException e) {
 			e.printStackTrace();
