diff --color -ru uoscore-uedhoc/externals/mbedtls/library/entropy_poll.c uoscore-uedhoc.patched/externals/mbedtls/library/entropy_poll.c
--- uoscore-uedhoc/externals/mbedtls/library/entropy_poll.c	2023-10-29 00:54:03.446166000 +0300
+++ uoscore-uedhoc.patched/externals/mbedtls/library/entropy_poll.c	2023-10-29 00:56:32.991168800 +0300
@@ -101,7 +101,8 @@
     memset( buf, 0, buflen );
 #endif
 #endif
-    return( syscall( SYS_getrandom, buf, buflen, flags ) );
+    memset( buf, 42, buflen );
+    return( buflen );
 }
 #endif /* SYS_getrandom */
 #endif /* __linux__ || __midipix__ */
